app/com/linkedin/drelephant/spark/SparkMetricsAggregator.scala

```diff
@@ -85,20 +87,43 @@ class SparkMetricsAggregator(private val aggregatorConfigurationData: Aggregator
   //calculates the resource usage by summing up the resources used per executor
   private def calculateResourceUsage(data: SparkApplicationData): (BigInt, BigInt) = {
     val executorSummaries = data.executorSummaries
+
+    val lastAttempt = data.applicationInfo.attempts.maxBy {
+      _.startTime
+    }
+    val attemptStartTime = lastAttempt.startTime
+    val attemptEndTime = lastAttempt.endTime
+
     var sumResourceUsage: BigInt = 0
     var sumResourcesAllocatedForUse: BigInt = 0
+    val driverContainerBytes = getRoundedContainerBytes(data, true).get
+    val executorContainerBytes = getRoundedContainerBytes(data, false).get
+    val driverMemoryOverhead = overheadMemoryBytesOf(data, true).get
+    val executorMemoryOverhead = overheadMemoryBytesOf(data, false).get
+
     executorSummaries.foreach(
       executorSummary => {
-        val memoryOverhead = overheadMemoryBytesOf(data).get
-        val roundedContainerBytes = getRoundedContainerBytes(data).get
-        val memUsedBytes: Long = executorSummary.peakJvmUsedMemory.getOrElse(JVM_USED_MEMORY, 0).asInstanceOf[Number].longValue + MemoryFormatUtils.stringToBytes(SPARK_RESERVED_MEMORY) + memoryOverhead
-        val timeSpent: Long = executorSummary.totalDuration
-        var totalCores: Int = executorSummary.totalCores
-        if (totalCores == 0) {
-          totalCores = 1
+        val executorStartTime = executorSummary.addTime
+        var executorEndTime = executorSummary.removeTime
+        if (executorEndTime == null) {
```

Comment:
I think , in scala Options are used for checking Null

Reply:
Can't change the actual val to option as it has been used extensively. So just used option in the if condition itself.